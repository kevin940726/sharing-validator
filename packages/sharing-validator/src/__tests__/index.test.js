import sharingValidator from "../";

test("kaihao.dev baseline", async () => {
  const results = await sharingValidator("https://kaihao.dev");

  expect(results).toMatchSnapshot();
});

test("kaihao.dev enable all features", async () => {
  const results = await sharingValidator("https://kaihao.dev", {
    facebook: true,
    twitter: true,
    AASA: true,
    assetlinks: true,
    facebookAppLinkIOS: true,
    facebookAppLinkAndroid: true
  });

  expect(results).toMatchSnapshot();
});
