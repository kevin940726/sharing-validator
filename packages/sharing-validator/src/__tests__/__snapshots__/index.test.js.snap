// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kaihao.dev baseline 1`] = `
Object {
  "meta": Object {
    "og": Object {
      "description": "I&#x27;m Kai Hao, a front-end developer in Taiwan.",
      "image": Object {
        "height": "626",
        "url": "https://kaihao.dev/static/15baf5b2b73048942a827496d3805470/70e2c/default-meta-image.png",
        "width": "1200",
      },
      "title": "Kai Hao",
      "type": "website",
      "url": "https://kaihao.dev/",
    },
  },
  "results": Object {
    "facebook": Object {
      "errors": Array [
        Object {
          "content": undefined,
          "message": "Expected to match /.+/, but received undefined.",
          "property": "fb:app_id",
          "valid": false,
        },
      ],
      "validations": Array [
        Object {
          "content": undefined,
          "message": "Expected to match /.+/, but received undefined.",
          "property": "fb:app_id",
          "valid": false,
        },
        Object {
          "content": "I&#x27;m Kai Hao, a front-end developer in Taiwan.",
          "message": "Expected to match /.+/, but received I&#x27;m Kai Hao, a front-end developer in Taiwan..",
          "property": "og:description",
          "valid": true,
        },
        Object {
          "content": "https://kaihao.dev/static/15baf5b2b73048942a827496d3805470/70e2c/default-meta-image.png",
          "message": "Expected to match /.+/, but received https://kaihao.dev/static/15baf5b2b73048942a827496d3805470/70e2c/default-meta-image.png.",
          "property": "og:image:url",
          "valid": true,
        },
        Object {
          "content": "1200",
          "message": "Expected \\"og:image:width\\" to be an integer >= 600, received 1200",
          "property": "og:image:width",
          "valid": true,
        },
        Object {
          "content": "626",
          "message": "Required to be integer, received 626.",
          "property": "og:image:height",
          "valid": true,
        },
      ],
    },
    "og": Object {
      "errors": Array [
        Object {
          "content": undefined,
          "message": "\\"og:image.alt\\" is required when specifying \\"og:image\\" or \\"og:image:url\\"",
          "property": "og:image:alt",
          "valid": false,
        },
      ],
      "validations": Array [
        Object {
          "content": "https://kaihao.dev/",
          "message": "Expected to match /.+/, but received https://kaihao.dev/.",
          "property": "og:url",
          "valid": true,
        },
        Object {
          "content": "Kai Hao",
          "message": "Expected to match /.+/, but received Kai Hao.",
          "property": "og:title",
          "valid": true,
        },
        Object {
          "content": "website",
          "message": "Expected to match /^(music\\\\.(song|album|playlist|radio_station)|video\\\\.(movie|episode|tv_show|other)|article|book|profile|website)$/, but received website.",
          "property": "og:type",
          "valid": true,
        },
        Object {
          "content": "I&#x27;m Kai Hao, a front-end developer in Taiwan.",
          "message": "Expected string, received I&#x27;m Kai Hao, a front-end developer in Taiwan..",
          "property": "og:description",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected string, received undefined.",
          "property": "og:site_name",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected determiner to be one of (a, an, the, \\"\\", auto), received undefined",
          "property": "og:determiner",
          "valid": true,
        },
        Object {
          "content": "https://kaihao.dev/static/15baf5b2b73048942a827496d3805470/70e2c/default-meta-image.png",
          "message": "Expected url to have 'http' or 'https' protocol, received https:",
          "property": "og:image:url",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:image:secure_url",
          "valid": true,
        },
        Object {
          "content": "1200",
          "message": "Expected integer, received 1200",
          "property": "og:image:width",
          "valid": true,
        },
        Object {
          "content": "626",
          "message": "Expected integer, received 626",
          "property": "og:image:height",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "\\"og:image.alt\\" is required when specifying \\"og:image\\" or \\"og:image:url\\"",
          "property": "og:image:alt",
          "valid": false,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:video:url",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:video:secure_url",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected integer, received undefined",
          "property": "og:video:width",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected integer, received undefined",
          "property": "og:video:height",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:audio:url",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:audio:secure_url",
          "valid": true,
        },
      ],
    },
    "twitter": Object {
      "errors": Array [
        Object {
          "content": undefined,
          "message": "Expected to match /^(summary|summary_large_image|app|player)$/, but received undefined.",
          "property": "twitter:card",
          "valid": false,
        },
      ],
      "validations": Array [
        Object {
          "content": undefined,
          "message": "Expected to match /^(summary|summary_large_image|app|player)$/, but received undefined.",
          "property": "twitter:card",
          "valid": false,
        },
        Object {
          "content": undefined,
          "message": "Expected to match /^@.+$|^$/, but received undefined.",
          "property": "twitter:site",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected to match /^@.+$|^$/, but received undefined.",
          "property": "twitter:creator",
          "valid": true,
        },
      ],
    },
  },
}
`;

exports[`kaihao.dev enable all features 1`] = `
Object {
  "meta": Object {
    "og": Object {
      "description": "I&#x27;m Kai Hao, a front-end developer in Taiwan.",
      "image": Object {
        "height": "626",
        "url": "https://kaihao.dev/static/15baf5b2b73048942a827496d3805470/70e2c/default-meta-image.png",
        "width": "1200",
      },
      "title": "Kai Hao",
      "type": "website",
      "url": "https://kaihao.dev/",
    },
  },
  "results": Object {
    "AASA": Object {
      "errors": Array [
        Object {
          "message": "Response status returns 404 when trying to access https://kaihao.dev/apple-app-site-association,
and returns status 404 when trying to access https://kaihao.dev/.well-known/apple-app-site-association.
One of the above url should be set up to be accessed.",
          "property": "apple-app-site-association",
          "valid": false,
        },
      ],
    },
    "facebook": Object {
      "errors": Array [
        Object {
          "content": undefined,
          "message": "Expected to match /.+/, but received undefined.",
          "property": "fb:app_id",
          "valid": false,
        },
      ],
      "validations": Array [
        Object {
          "content": undefined,
          "message": "Expected to match /.+/, but received undefined.",
          "property": "fb:app_id",
          "valid": false,
        },
        Object {
          "content": "I&#x27;m Kai Hao, a front-end developer in Taiwan.",
          "message": "Expected to match /.+/, but received I&#x27;m Kai Hao, a front-end developer in Taiwan..",
          "property": "og:description",
          "valid": true,
        },
        Object {
          "content": "https://kaihao.dev/static/15baf5b2b73048942a827496d3805470/70e2c/default-meta-image.png",
          "message": "Expected to match /.+/, but received https://kaihao.dev/static/15baf5b2b73048942a827496d3805470/70e2c/default-meta-image.png.",
          "property": "og:image:url",
          "valid": true,
        },
        Object {
          "content": "1200",
          "message": "Expected \\"og:image:width\\" to be an integer >= 600, received 1200",
          "property": "og:image:width",
          "valid": true,
        },
        Object {
          "content": "626",
          "message": "Required to be integer, received 626.",
          "property": "og:image:height",
          "valid": true,
        },
      ],
    },
    "facebookAppLink": Object {
      "android": Object {
        "errors": Array [
          Object {
            "content": undefined,
            "message": "Expected to match /.+/, but received undefined.",
            "property": "al:android:app_name",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /.+/, but received undefined.",
            "property": "al:android:package",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /^.+:\\\\/\\\\/.*$/, but received undefined.",
            "property": "al:android:url",
            "valid": false,
          },
        ],
        "validations": Array [
          Object {
            "content": undefined,
            "message": "Expected to match /.+/, but received undefined.",
            "property": "al:android:app_name",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /.+/, but received undefined.",
            "property": "al:android:package",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /^.+:\\\\/\\\\/.*$/, but received undefined.",
            "property": "al:android:url",
            "valid": false,
          },
        ],
      },
      "ios": Object {
        "errors": Array [
          Object {
            "content": undefined,
            "message": "Expected to match /.+/, but received undefined.",
            "property": "al:ios:app_name",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /.+/, but received undefined.",
            "property": "al:ios:app_store_id",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /^.+:\\\\/\\\\/.*$/, but received undefined.",
            "property": "al:ios:url",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /^false$/, but received undefined.",
            "property": "al:web:should_fallback",
            "valid": false,
          },
        ],
        "validations": Array [
          Object {
            "content": undefined,
            "message": "Expected to match /.+/, but received undefined.",
            "property": "al:ios:app_name",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /.+/, but received undefined.",
            "property": "al:ios:app_store_id",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /^.+:\\\\/\\\\/.*$/, but received undefined.",
            "property": "al:ios:url",
            "valid": false,
          },
          Object {
            "content": undefined,
            "message": "Expected to match /^false$/, but received undefined.",
            "property": "al:web:should_fallback",
            "valid": false,
          },
        ],
      },
    },
    "og": Object {
      "errors": Array [
        Object {
          "content": undefined,
          "message": "\\"og:image.alt\\" is required when specifying \\"og:image\\" or \\"og:image:url\\"",
          "property": "og:image:alt",
          "valid": false,
        },
      ],
      "validations": Array [
        Object {
          "content": "https://kaihao.dev/",
          "message": "Expected to match /.+/, but received https://kaihao.dev/.",
          "property": "og:url",
          "valid": true,
        },
        Object {
          "content": "Kai Hao",
          "message": "Expected to match /.+/, but received Kai Hao.",
          "property": "og:title",
          "valid": true,
        },
        Object {
          "content": "website",
          "message": "Expected to match /^(music\\\\.(song|album|playlist|radio_station)|video\\\\.(movie|episode|tv_show|other)|article|book|profile|website)$/, but received website.",
          "property": "og:type",
          "valid": true,
        },
        Object {
          "content": "I&#x27;m Kai Hao, a front-end developer in Taiwan.",
          "message": "Expected string, received I&#x27;m Kai Hao, a front-end developer in Taiwan..",
          "property": "og:description",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected string, received undefined.",
          "property": "og:site_name",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected determiner to be one of (a, an, the, \\"\\", auto), received undefined",
          "property": "og:determiner",
          "valid": true,
        },
        Object {
          "content": "https://kaihao.dev/static/15baf5b2b73048942a827496d3805470/70e2c/default-meta-image.png",
          "message": "Expected url to have 'http' or 'https' protocol, received https:",
          "property": "og:image:url",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:image:secure_url",
          "valid": true,
        },
        Object {
          "content": "1200",
          "message": "Expected integer, received 1200",
          "property": "og:image:width",
          "valid": true,
        },
        Object {
          "content": "626",
          "message": "Expected integer, received 626",
          "property": "og:image:height",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "\\"og:image.alt\\" is required when specifying \\"og:image\\" or \\"og:image:url\\"",
          "property": "og:image:alt",
          "valid": false,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:video:url",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:video:secure_url",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected integer, received undefined",
          "property": "og:video:width",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected integer, received undefined",
          "property": "og:video:height",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:audio:url",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Failed to parse the url. Invalid URL",
          "property": "og:audio:secure_url",
          "valid": true,
        },
      ],
    },
    "twitter": Object {
      "errors": Array [
        Object {
          "content": undefined,
          "message": "Expected to match /^(summary|summary_large_image|app|player)$/, but received undefined.",
          "property": "twitter:card",
          "valid": false,
        },
      ],
      "validations": Array [
        Object {
          "content": undefined,
          "message": "Expected to match /^(summary|summary_large_image|app|player)$/, but received undefined.",
          "property": "twitter:card",
          "valid": false,
        },
        Object {
          "content": undefined,
          "message": "Expected to match /^@.+$|^$/, but received undefined.",
          "property": "twitter:site",
          "valid": true,
        },
        Object {
          "content": undefined,
          "message": "Expected to match /^@.+$|^$/, but received undefined.",
          "property": "twitter:creator",
          "valid": true,
        },
      ],
    },
  },
}
`;
